File: python/code_examples/09-equality-truthiness/main.js
// === strict equality (no coercion); == coerces types
// truthy/falsy: "", 0, null, undefined, NaN are falsy

function main() {
  console.log(Boolean([]), Boolean([1])); // true true (empty array is truthy!)
  console.log(1 == true, 1 === true); // true false
  console.log("" == false); // true (coercion)
}

main();

File: python/code_examples/09-equality-truthiness/main.py
# == compares value; custom types can override __eq__
# "truthy": empty containers are falsy ([], {}, "", 0, None)

def main() -> None:
    print(bool([]), bool([1]))  # False True
    print(1 == True)  # False in Python (int vs bool)
    print(1 == 1.0)  # True


if __name__ == "__main__":
    main()
